### PR TITLE
Force clients to connect with SSL

### PR DESCRIPTION
**Pull request by NyakudyaA**
Fixes https://dba.stackexchange.com/questions/8580/force-postgresql-clients-to-use-ssl/8622#8622

### CODE REVIEW

File: scripts/setup-pg_hba.sh [hunk 1]
```diff
@@ -12,24 +12,35 @@ fi
 # Reconfigure pg_hba if environment settings changed
 cat ${ROOT_CONF}/pg_hba.conf.template > ${ROOT_CONF}/pg_hba.conf
 
+if [[ "$FORCE_SSL" =~ [Tt][Rr][Uu][Ee] ]]; then
+  PG_CONF_HOST='hostssl'
+  CERT_AUTH='cert'
```
Comment (lucernae) [inline scripts/setup-pg_hba.sh:17]: Doesn't cert auth uses full verified certificate chain?
How are we going to test this?

**Review by lucernae — COMMENTED** [review]
Can we push this branch to kartoza/postgis and make PR from there so that Github Action generate a docker image for the current PR

File: scripts/setup-pg_hba.sh (hunk 1 above)
Comment (NyakudyaA) [reply scripts/setup-pg_hba.sh:17]: You can connect the service and call `psql "dbname=gis port=5432 user=docker host=localhost sslmode=verify-full sslcert=/etc/ssl/certs/ssl-cert-snakeoil.pem sslkey=/etc/ssl/private/ssl-cert-snakeoil.key sslrootcert=<path>/root.crt"

Comment (NyakudyaA) [reply scripts/setup-pg_hba.sh:17]: full verified is client side option not server side

**Review by timlinux — COMMENTED** [review]
Thanks @NyakudyaA. Do I read it right that ssl wont be enforced by default? Do you think we will break many applications if we make it so that you have to explicitly disable ssl for clients rather than explicityly enabled it?

**Comment by NyakudyaA** [comment]
> Thanks @NyakudyaA. Do I read it right that ssl wont be enforced by default? Do you think we will break many applications if we make it so that you have to explicitly disable ssl for clients rather than explicityly enabled it?

Yes, It should be disabled by default because it will break all installations that are already running. We can only enable this by default in PostgreSQL 14

**Comment by timlinux** [comment]
Ok sounds good LGTM

**Pull request merged**